### ISSUE TITLE
Installing t-tabular requires out-of-band knowledge about additional dependencies

### ISSUE THREAD

**Issue by jindrichmynarz** [UnifiedViews/Plugins#151]
When you want to install t-tabular (checked with develop branch, commit ebba52f83f47bc7303fbe9f07524fc03eaf918f9), you need to know that this DPU depends on additional libraries that need to be provided in `{module.path}/lib`. More precisely, the DPU requires org.apache.poi-bundle-3.10.0.jar (can be found [here](https://github.com/mff-uk/DPUs/tree/master/dpu/t-tabular/libs)) and org.junit-bundle-4.11.0.jar (can be found [here](https://github.com/mff-uk/DPUs/tree/master/dependencies)), which are not built during the compilation of the UnifiedViews Core.

At minimum, this requirement should be explicitly documented. However, it would be better if this DPU worked with a vanilla installation of UnifiedViews, i.e. not depending on bundles that UnifiedViews doesn't provide by default (either by removing the dependency or making UnifiedViews/Core build these dependencies).

**Comment by jindrichmynarz** [comment]
As discussed today, this might be solved in the context of the more general issue, which is how to structure documentation of DPUs.

**Comment by tomas-knap** [comment]
This should be in the developer's notes in README for the respective DPU. 

a) Developers will not update any wiki, b) wiki may be generated from README files

**Comment by jindrichmynarz** [comment]
Then I think the developer of t-tabular should be assigned on this issue.

**Comment by tomas-knap** [comment]
@skodape Please update README and also copy all the required external libs to t-tabular/libs (currently available at: 

* https://github.com/mff-uk/DPUs/tree/master/dpu/t-tabular/libs
* https://github.com/mff-uk/DPUs/tree/master/dependencies

**Comment by skrchnavy** [comment]
@tomas-knap is it OK to have these libs in github? Doesn't maven provide support for these libraries?

**Comment by tomas-knap** [comment]
One of the library is not on mavem. Further, the library has to anyway be put to target/lib.

**Comment by skrchnavy** [comment]
@tomas-knap I will check how such external dependencies shall be treated in maven environment

**Comment by skrchnavy** [comment]
@tomas-knap @skodape there are few ways how to support dependencies (it is not good practice to keep binary files in the git repo together with sources due to size of repo when versions are updated):
* github maven repo  (http://kwebble.com/blog/2014/02/19/use-github-to-host-your-own-maven-repo)
* to use local repo (https://maven.apache.org/guides/mini/guide-configuring-maven.html) where you can manually add missing jars
* to publish jar file into an available repo (maven.eea.sk is ready for this).

**Comment by tomas-knap** [comment]
I would suggest to use github maven repo (as this is public repo, associated with github project) 

But it is not a crucial adjustment, it does not need to be done in 2.0

**Comment by tomas-knap** [comment]
Please initialize github maven repo as introduced here: github maven repo (http://kwebble.com/blog/2014/02/19/use-github-to-host-your-own-maven-repo). And put the two libraries there. Then we will close this issue

**Comment by skrchnavy** [comment]
@skodape / @tomas-knap : is there a difference between https://github.com/mff-uk/DPUs/blob/master/dependencies/org.junit-bundle-4.11.0.jar and http://mvnrepository.com/artifact/org.apache.servicemix.bundles/org.apache.servicemix.bundles.junit?

**Comment by skrchnavy** [comment]
github maven repo initialized (https://github.com/UnifiedViews/Maven-Repo)
```xml
  <repository>
    <id>UnifiedViews_Maven_Repo</id>
    <name>UnifiedViews Git based maven repo</name>
    <url>https://github.com/UnifiedViews/Maven-Repo/tree/master/</url>
  </repository>
```

 and poi-bundle added into repo so dependency is:
```xml
  <dependency>
    <groupId>org.apache.poi</groupId>
    <artifactId>poi-bundle</artifactId>
    <version>3.10.0</version>
  </dependency>
```

**Comment by mkl-eea-sk** [comment]
@skodape Have you  been creating OSGI bundles on your own or you have downloaded them somewhere? And from where.

**Comment by mkl-eea-sk** [comment]
https://poi.apache.org/changes.html Version 3.10-FINAL (2014-02-08)

**Comment by mkl-eea-sk** [comment]
@jindrichmynarz can you please help me, by providing export of any pipeline which tests t-tabular thorougly? xls(or xslx) especially! (csv and dbf should work).
or just give me such xls and screenshot of t-tabular configuration. i will do initial testing.
(or if you are brave enough, i may provide feature branch which you can test whether it fixes this issue and t-tabular still works :) )

**Comment by skrchnavy** [comment]
Warning: please do not put any commit related to this issue into develop nor feature finish prior to release branch 2.1.0 created. This is planed for 2.2.0

**Comment by mkl-eea-sk** [comment]
Understood (this applies to all milestone 2.2.0 issues)

**Comment by mkl-eea-sk** [comment]
@jindrichmynarz you can test, if you want, code from this branch https://github.com/UnifiedViews/Plugins/tree/feature/resolve%23151
it works for me (and pipeline SK-60) i have got from you (remove t-findAndReplaceInFIles).
It handles XLS, not XLSX, nor ODS. But t-tabular before my changes was unable to handle them too.

**Comment by mkl-eea-sk** [comment]
Useful link http://apache-poi.1045710.n5.nabble.com/Bug-57857-New-Provide-an-OSGi-bundle-tp5718581p5718588.html

**Comment by mkl-eea-sk** [comment]
Resolved also by this pull request https://github.com/UnifiedViews/Core/pull/440

**Comment by skrchnavy** [comment]
Reopened, need to be tested how DPU 2.1.1 (SNAPSHOT) works with UV 2.1.0 and also if T-Tabular 2.1.0 works with core 2.1.1 (SNAPSHOT)

**Comment by mkl-eea-sk** [comment]
both situations are matter of documentation. old tabular on new core -> remove these libs, add these. new tabular on old core -> add these libs. old tabular on old core -> add these (another) libs.
new tabular on new core - just works.

----- Original Message -----

**Comment by skrchnavy** [comment]
@tomas-knap / @jakubklimek  please be aware of this issue, it is included in 2.1.1 release so when you observe issue with t-tabular you have to remove some jars from lib dir or to copy proper one.
@mkl-eea-sk please provide list of jars for combinations that shall be installed manually:
UV/Core 2.1.0 / t-tabular@Plugins 2.1.0:
UV/Core 2.1.1 / t-tabular@Plugions 2.1.0:
UV/Core 2.1.0 / t-tabular@Plugins 2.1.1 (SNAPSHOT):
UV/Core 2.1.1 / t-tabular@Plugins 2.1.1 (SNAPSHOT):
